### PR TITLE
[MB-14381] setFinancialReviewFlag refactor and now capturing remarks

### PR DESCRIPTION
**Pull request by ruizajtruss**
## Summary

Usual refactor however it is now capturing remarks as well 🎉 

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the UI locally.

```sh
make client_run
```

##### Terminal 2

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

1. Login as an office user
2. In the move details tab; flag a shipment for financial review
3. You should now see the details in the move history tab

## Verification Steps for Author

These are to be checked by the author.

- [x] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

## Screenshots
Before: 
Screenshot 2022-10-27 at 4 44 25 PM

After:
Screenshot 2022-10-27 at 4 46 02 PM

### CODE REVIEW

**Comment by mmayo-truss** [comment]
This is working as expected. 
image

File: src/constants/MoveHistory/EventTemplates/SetFinancialReviewFlag/setFinancialReviewFlag.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import LabeledDetails from 'pages/Office/MoveHistory/LabeledDetails';
+
+export default {
+  action: a.UPDATE,
+  eventName: o.setFinancialReviewFlag,
+  tableName: t.moves,
+  getEventNameDisplay: () => 'Flagged move',
+  getDetails: (historyRecord) => (
+    <>
+      {historyRecord.changedValues?.financial_review_flag === 'true'
```
Comment (mmayo-truss) [inline src/constants/MoveHistory/EventTemplates/SetFinancialReviewFlag/setFinancialReviewFlag.jsx:15]: Praise: This is a good example of the flexibility we have with these refactors! Good work

File: src/constants/MoveHistory/EventTemplates/SetFinancialReviewFlag/setFinancialReviewFlag.test.jsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    },
+    tableName: 'moves',
+  };
+
+  it('correctly matches the Set financial review flag event', () => {
+    const template = getTemplate(historyRecord);
+    expect(template).toMatchObject(e);
+  });
+
+  it('correctly displays changed values in details column', () => {
```
Comment (mmayo-truss) [inline src/constants/MoveHistory/EventTemplates/SetFinancialReviewFlag/setFinancialReviewFlag.test.jsx:22]: Praise: I like how you broke this out into three separate tests.

**Review by juliemoon — COMMENTED** [review]
flagged move screenshot
@ruizajtruss this works great! I was able to test the functionality and I can see the remarks and the "moved flagged for financial review". One thing I noticed is under the event column it should say **Updated shipment** not Flagged move.

**Comment by ruizajtruss** [comment]
> This is working as expected, though I don't see a way to add remarks when we are unflagging. image
> 
> Question: Do we want to hide the remarks if there is no information?

Odd, the input field should allow you to modify the remark when unflagging 🤔

**Comment by mmayo-truss** [comment]
> > Question: Do we want to hide the remarks if there is no information?
> 
> I figure we wouldn't want to hide it since I think it's more in line with the other remarks where we display a double dash when nothing is present.

That makes sense to me.

**Review by mmayo-truss — APPROVED** [review]
LGTM. 🏅

**Comment by ruizajtruss** [comment]
> flagged move screenshot
> 
> @ruizajtruss this works great! I was able to test the functionality and I can see the remarks and the "moved flagged for financial review". One thing I noticed is under the event column it should say **Updated shipment** not Flagged move.



> flagged move screenshot
> 
> @ruizajtruss this works great! I was able to test the functionality and I can see the remarks and the "moved flagged for financial review". One thing I noticed is under the event column it should say **Updated shipment** not Flagged move.

Screenshot 2022-10-28 at 10 55 04 AM
 Done!

**Pull request merged**